game/controllers-joystick.cc

```diff
@@ -20,25 +20,29 @@ namespace input {
 				m_joysticks.push_back(joy);
 			}
 		}
-		std::string getName(unsigned device) const override {
+		std::string getName(int device) const override {
 			return SDL_JoystickNameForIndex(device);
 		}
 		bool process(Event& event, SDL_Event const& sdlEv) override {
 			if (sdlEv.type == SDL_JOYBUTTONDOWN || sdlEv.type == SDL_JOYBUTTONUP) {
 				event.hw = sdlEv.jbutton.button;
-				event.value = (sdlEv.type == SDL_JOYBUTTONDOWN ? 1.0 : 0.0);
```

Comment:
careful, this code is not equivalent, the event.value is not set when type != SDL_JOYBUTTONDOWN now

Reply:
I want to look well into this because there is a joystick bug I haven't been able to get yet.

However, I don't think I'm ever actually checking the optional value directly, but rather using a function that returns `.value_or(0.0)`